src/accelerate/state.py

```diff
@@ -179,7 +179,10 @@ def __init__(self, cpu: bool = False, **kwargs):
                 )
 
             # Sets up self.backend + imports
-            backend, distributed_type = self._prepare_backend(cpu, use_sagemaker_dp, kwargs.pop("backend", None))
+            original_backend = kwargs.pop("backend", None)
+            backend, distributed_type = self._prepare_backend(cpu, use_sagemaker_dp, original_backend)
+            if original_backend and backend != original_backend:
```

Comment:
One last final check, sorry missed this the first time. Let's not trust python here 😉 
```suggestion
            if original_backend is not None and backend != original_backend:
```

Reply:
Done.